src/cpu.rs

```diff
@@ -63,8 +71,109 @@ impl CPU {
 
                 let data_start = self.registers.program_counter.wrapping_add(1);
 
-                let slice = self.memory.get_slice(data_start, extra_bytes);
-                let am_out = am.process(self, slice);
+                let slice = if extra_bytes == 0 {
+                    [0, 0]
+                } else if extra_bytes == 1 {
+                    [self.memory.get_byte(data_start), 0]
+                } else if extra_bytes == 2 {
+                    [
+                        self.memory.get_byte(data_start),
+                        self.memory.get_byte(data_start.wrapping_add(1)),
+                    ]
+                } else {
+                    panic!()
```

Comment:
We could add an error message here.
```suggestion
                    panic!("CPU cannot read more than 2 extra bytes from memory at a time")
```

Reply:
Yeah, we can leave it out. For nostd we'd need our own panic handler anyway, so maybe an [`unreachable_unchecked()`](https://doc.rust-lang.org/std/hint/fn.unreachable_unchecked.html) would be an alternative. I don't mind either way. 😉